src/libs/conduit/conduit_utils.hpp

```diff
@@ -238,6 +238,76 @@ namespace utils
                                           const std::string &file,
                                           int line);
 
+
+//-----------------------------------------------------------------------------
+/// Primary interface used by the conduit API to move memory.
+//-----------------------------------------------------------------------------
+
+    // conduit uses a single pair of memset and memcpy functions to
+    // manage data movement. 
+
+    // this strategy allows downstream users to support complex cases
+    // like moving between memory spaces non accessible on the host. 
+    //
+    // These methods aren't bound to allocators b/c allocators
+    // won't be tied into all of the places where source and dest pointers
+    // need to be located.
```

Comment:
The first pass had these pinned to allocators. 

The problem is while a Node may know its allocator (and it may even know the allocator of a Node that is being set into it) there is no good notion of how to move data between allocators. Umpire provides a platform interface to solve this, but we don't want to go that far.  There are also cases where allocator is completely unknown and those are most of the interface. 
 
So -- In a bunch of the cases where we need to move data, the allocator (or more broadly where the data lives), both source or dest, won't be known explicitly by any tagging or metadata that conduit has access to. The only solution is pointer inspection. The solve this --- folks doing fancy things will create a memcpy that uses pointer inspection to handle all situations in play. So the one registered instance will have to be smart enough to move between any funky mem spaces in use. Matt is doing this in Ascent, so that will be a concrete example to test these ideas.

Reply:
Note: printing a node containing device values segfaults, but I didn't expect that to work.

Reply:
perhaps `bool calamitous()`

Reply:
What also doesn't work (which I expected) is setting a gpu allocator to a node and passing it to the blueprint examples. I don't think we need to handle this case, because its easy enough to `Node device_node = host_node;`